Modules/_testcapi/heaptype_relative.c

```diff
@@ -1,4 +1,11 @@
+#ifndef _MSC_VER
+#include "pyconfig.h"   // Py_NOGIL
```

Comment:
Hmm IIUC, parts.h includes python.h and python.h includes pyconfig.h
I think that @vstinner can left the comment for this.
https://github.com/python/cpython/blob/677d4bc15e999aa57abe7d23d7fac2c34a6fe0ba/Include/Python.h#L12-L15

Reply:
Yeah, but I also think it is fine to include it explicitly. OTOH, it would make the diff smaller , FWIW.

Why is it only included on Windows?

Reply:
* We can't define `Py_LIMITED_API` in `--disable-gil` builds because the required atomic operations are not available in the limited API and `--disable-gil` builds are not compatible with the stable ABI. I expect this to change in the future -- there was a productive discussion at the sprint about this, but it'll require a PEP and take a while.
* So we need to guard the `#define Py_LIMITED_API 0x030...` with a `#ifndef Py_NOGIL` for now
* On `configure` based systems (Linux and macOS), `Py_NOGIL` is defined in `pyconfig.h`, so we need to include that before setting `Py_LIMITED_API`. We can't include all of `"Python.h"` because that definition depends on `Py_LIMITED_API`
* The situation is different on Windows. On Windows, `Py_NOGIL` is passed as a build flag and the Windows [`pyconfig.h`](https://github.com/python/cpython/blob/11312eae6ec3acf51aacafce4cb6d1a5edfd5f2e/PC/pyconfig.h#L312) depends on `Py_LIMITED_API` so we should *not* include it before setting `Py_LIMITED_API`.

Reply:
Ok, thanks for the details :)